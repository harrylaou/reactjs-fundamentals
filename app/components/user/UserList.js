import React, {Component} from 'react'
import {List, ListItem} from 'material-ui/List'
import ActionGrade from 'material-ui/svg-icons/action/grade'
import Avatar from 'material-ui/Avatar'
import {pinkA200} from 'material-ui/styles/colors'
import withWidth, { LARGE} from '../../utils/WithWidth'


const UserList = (props) => {
  let listItems = props.users.map(user => (
    <ListItem
      onClick={props.showUserProfile.bind(this, user)} key={user.username} style={{color: "black"}}
      primaryText={ `${user.name.first} ${user.name.last}`}
      leftIcon={<ActionGrade color={pinkA200} />}
      rightAvatar={<Avatar src={`images/${user.username}_sm.jpg`} />}
    />
  ))

  const listView = <List className="view">{listItems}</List>
  const childrenView = <div className="view">{props.children}</div>

  let masterView, detailView

  if (props.username && props.width !== LARGE) {
    masterView = childrenView
    detailView = null
  } else {
    masterView = listView
    detailView = childrenView
  }
  return (
    <div className="view-container">
      {masterView}
      {detailView}
    </div>
  )
}

export default withWidth()(UserList)
