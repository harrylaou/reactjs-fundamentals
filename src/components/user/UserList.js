import React, { Component } from 'react'
import { List, ListItem } from 'material-ui/List'
import ActionGrade from 'material-ui/svg-icons/action/grade'
import Avatar from 'material-ui/Avatar'
import { pinkA200 } from 'material-ui/styles/colors'
import { LARGE } from '../../utils/WithWidth'

const UserList = (props) => {
  let listItems = props.users.map(user => (
    <ListItem
      onClick={props.showUserProfile.bind(this, user)} key={user.username} style={{color: "black"}}
      primaryText={ `${user.name.first} ${user.name.last}`}
      leftIcon={<ActionGrade color={pinkA200} />}
      rightAvatar={<Avatar src={`images/${user.username}_sm.jpg`} />}
    />
  ))

  const listView = <List className="view">{listItems}</List>
  const childrenView = <div className="view">{props.children}</div>

  let masterView, detailView

  if (props.username && props.width !== LARGE) {
    masterView = childrenView
    detailView = null
  } else {
    masterView = listView
    detailView = childrenView
  }
  return (
    <div className="view-container">
      {masterView}
      {detailView}
    </div>
  )
}

UserList.propTypes = {
  users: React.PropTypes.array.isRequired,
  username: React.PropTypes.string,
  width: React.PropTypes.number.isRequired
}

export default UserList
