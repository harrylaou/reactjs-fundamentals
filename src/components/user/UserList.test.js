import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import UserList from './UserList'

describe('<UserList />', () => {
  it('should render with default props', () => {
    const onClick = jest.fn()
    const user = {
      username: 'paco',
      name: {
        first: 'paco',
        last: 'martinez'
      }
    }
    const props = {
      users: [user],
      width: 3,
      showUserProfile: onClick
    }

    const wrapper = shallow(
      <UserList {...props} />
    )

    expect(shallowToJson(wrapper)).toMatchSnapshot()

    wrapper.find('ListItem').simulate('click')

    expect(onClick).toBeCalledWith(user)
  })

  it('should not render detailview', () => {
    const user = {
      username: 'paco',
      name: {
        first: 'paco',
        last: 'martinez'
      }
    }
    const props = {
      username: 'paco',
      users: [user],
      width: 1,
      showUserProfile: () => {}
    }

    const wrapper = shallow(
      <UserList {...props} />
    )

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})