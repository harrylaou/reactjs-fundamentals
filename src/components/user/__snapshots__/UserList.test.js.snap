// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList /> should render with default props 1`] = `
<div
  className="view-container"
>
  <List
    className="view"
  >
    <ListItem
      autoGenerateNestedIndicator={true}
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      leftIcon={
        <ActionGrade
          color="#ff4081"
        />
      }
      nestedItems={Array []}
      nestedLevel={0}
      onClick={[Function]}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="paco martinez"
      primaryTogglesNestedList={false}
      rightAvatar={
        <Avatar
          size={40}
          src="images/paco_sm.jpg"
        />
      }
      secondaryTextLines={1}
      style={
        Object {
          "color": "black",
        }
      }
    />
  </List>
  <div
    className="view"
  />
</div>
`;
